GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -883,4 +885,24 @@ public GVREventManager getEventManager() {
     public GVRScriptManager getScriptManager() {
         return mScriptManager;
     }
+
+
+    protected long mGlDeleterPtr;
+
+    @Override
+    public void finalize() throws Throwable {
+        try {
+            if (0 != mGlDeleterPtr) {
```

Comment:
Yes, it is scattered as it came as an afterthought. This is what I could come up with quickly to eliminate at least some of the static state.

There must be a simple, more robust way but it needs thought. Like the gvrf finalizer thread managing the gl cleanup also. That whole thread-local stuff I don't like.